src/engine/CartesianProductJoin.cpp

```diff
@@ -7,6 +7,12 @@
 #include "engine/CallFixedSize.h"
 #include "util/Views.h"
 
+namespace {
+constexpr std::string_view recomputeMessage =
+    "After limits have been applied to the children, we can't compute the "
+    "results again in case the index changed.";
```

Comment:
I don't understand the `in case the index changed` part. What exactly is the issue here?

Reply:
+ it is also forbidden to clone, reflect that in the message.

Reply:
It's phrased confusingly. Basically if the data was static, we could just *not* apply the limit a second time, but since the results may change we can't do that.